### DISCUSSION TITLE
Query parameter for POST request

### DISCUSSION THREAD

**Discussion by ThuGapy** [PostgREST/postgrest#3557] [Q&A] [answered]
I am using PostgREST to do an HTTP integration with Chirpstack.

The way Chirpstack works is it does a POST requests with the body being the json data of the event, it also appends event=eventstatus to the URL. As far as I can understand, PostgREST tries to use the event query parameter as a filter, which then leads to the endpoint returning 400 (Bad request).

I was wondering if there was a way to fix this. I also tried to add the event parameter to my PL/pgSQL function as a parameter, but I think it's trying to fetch it from the body instead of the URL.

Also, I would not care if the event query parameter would simply be ignored.

Anything would be greatly appreciated.

**Comment by laurenceisla** [comment, marked as answer]
Currently there's no way to configure which query parameters to ignore directly in PostgREST. You'll need to use a reverse proxy like Nginx to handle those query parameters. See the docs for [an example on how to set Nginx for PostgREST](https://postgrest.org/en/v12/explanations/nginx.html#nginx).

In that example you can handle the query parameter at the beginning of the `location` definition, something like this:

```nginx

location /api/ {

  # This matches any query string that ends in `event=<any_value>`
  # And replaces it with what was before that query param
  if ($args ~* "(.*)(event=.*)") {
    set $args "$1";
  }

  # The rest of the configuration ...
} 
```

This the simplest attempt, it can be better refined to your needs. For example, since it's appended, I'm assuming that the query param goes at the end. Also, if the `event` param is intended (i.e. for a table that has an `event` column) then, specifying the possible eventstatus might be better, like: `(event=(eventstatus1|eventstatus2))`.

**Reply by ThuGapy** [reply]
Reverse proxy worked, but I'm not sure it's an ideal solution.

**Reply by patrickmichalina** [reply]
I have to do a similar thing with caddy in order to support unnamed JSON POST body + query params. This strikes me as something that should be supported natively in PostgREST.

**Reply by steve-chavez** [reply]
> This strikes me as something that should be supported natively in PostgREST.

A config with a list of globally ignored query parameters? Possible, but since PostgREST needs a proxy anyway I think it's cleaner to solve this there.

(I've also seen a similar case when PostgREST is behind `Kong`, which passes an [apikey query parameter](https://docs.konghq.com/hub/kong-inc/key-auth/#make-a-request-with-the-key))
